psi-sample/client/src/main/java/com/r3/conclave/sample/client/Client.java

```diff
@@ -2,130 +2,140 @@
 
 import com.esotericsoftware.kryo.Kryo;
 import com.esotericsoftware.kryo.io.Output;
-import com.r3.conclave.client.EnclaveConstraint;
-import com.r3.conclave.client.InvalidEnclaveException;
-import com.r3.conclave.common.EnclaveInstanceInfo;
-import com.r3.conclave.mail.Curve25519PrivateKey;
+import com.r3.conclave.client.EnclaveClient;
+import com.r3.conclave.client.web.WebEnclaveTransport;
+import com.r3.conclave.common.EnclaveConstraint;
 import com.r3.conclave.mail.EnclaveMail;
-import com.r3.conclave.mail.PostOffice;
 import com.r3.conclave.sample.common.AdDetails;
 import com.r3.conclave.sample.common.InputData;
 import com.r3.conclave.sample.common.InputDataSerializer;
 import com.r3.conclave.sample.common.UserDetails;
+import picocli.CommandLine;
 
-import java.io.*;
-import java.net.InetAddress;
-import java.net.InetSocketAddress;
-import java.net.Socket;
-import java.security.PrivateKey;
+import java.io.ByteArrayOutputStream;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.UUID;
+import java.util.concurrent.Callable;
 
 /**
  * Client can be of two types: Merchant or Service Provider
  * Merchant - who supplies list of users who have made a purchase
  * Service Providers - who supplies a list of users who have clicked on the ad
  * Both the clients send the lists to enclave, which calculates the ad conversion rate and sends it back to the clients
  */
-public class Client {
-    private static final String MERCHANT = "MERCHANT";
-    private static final String SERVICE_PROVIDER = "SERVICE-PROVIDER";
+@CommandLine.Command(name = "psi-client",
+        mixinStandardHelpOptions = true,
+        description = "Simple client that communicates with the PSIEnclave using the web host.")
+public class Client implements Callable<Void> {
+
+    @CommandLine.Option(names = {"-r", "--role"}
+            , description = "Role Options: ${MERCHANT/SERVICE_PROVIDER}")
+    Role role = null;
+
+    //use picocli to provide command line parameters
+    @CommandLine.Parameters( description = "List of credit card numbers to be sent to enclave")
+        private List<String> creditCardNumbers = new ArrayList<String>();
+
+    @CommandLine.Option(names = {"-u", "--url"},
+                required = true,
+                description = "URL of the web host running the enclave.")
+        private String url;
```

Comment:
You can use Intellij to auto indent and format your code.

Reply:
done